api/v1alpha1/clustertemplate_types.go

```diff
@@ -28,9 +28,25 @@ type ClusterTemplateSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// JSON formatted string that includes the schema for the ClusterTemplate.
-	// Required values should be explicitly defined.
-	InputDataSchema string `json:"inputDataSchema"`
+	InputDataSchema InputDataSchema `json:"inputDataSchema"`
```

Comment:
In the context of a hardware manager, a cluster template contains hardware templates with predefined configurations or initial values. The input data fields can then customize or override specific values as needed, rather than necessarily adhering to a schema that specifies how the JSON data should be structured.